pkg/pillar/cmd/zedrouter/probe.go

```diff
@@ -226,6 +219,19 @@ func checkNIprobeUplink(ctx *zedrouterContext, status *types.NetworkInstanceStat
 				break
 			}
 		}
+		// If none of the interfaces have valid IP address just pick the first
+		if status.CurrentUplinkIntf == "" {
+			if len(status.PInfo) > 0 {
+				var port string
+				for port = range status.PInfo {
+					break
+				}
+				info := status.PInfo[port]
```

Comment:
Do you want to check that it at least has a local IP address?

Reply:
Will do. We first check for unicast IP addresses. If not found check for atleast local IP addresses. If everything fails just pick the first port from list.